apps/stripe/src/modules/stripe/payment-intent-handling-results.ts

```diff
@@ -0,0 +1,167 @@
+import { buildSyncWebhookResponsePayload } from "@saleor/app-sdk/handlers/shared";
+
+import { BaseError } from "@/lib/errors";
+import { SaleorMoney } from "@/modules/saleor/saleor-money";
+import { StripePaymentIntentId } from "@/modules/stripe/stripe-payment-intent-id";
+
+import { StripePaymentIntentStatus } from "./stripe-payment-intent-status";
+
+abstract class SuccessResult {
+  readonly result = "" as "CHARGE_SUCCESS" | "AUTHORIZATION_SUCCESS";
+  readonly saleorMoney: SaleorMoney;
+  readonly stripePaymentIntentId: StripePaymentIntentId;
+
+  constructor(args: { saleorMoney: SaleorMoney; stripePaymentIntentId: StripePaymentIntentId }) {
+    this.saleorMoney = args.saleorMoney;
+    this.stripePaymentIntentId = args.stripePaymentIntentId;
+  }
+
+  getResponse() {
+    const typeSafeResponse = buildSyncWebhookResponsePayload<"TRANSACTION_PROCESS_SESSION">({
+      result: this.result,
+      amount: this.saleorMoney.amount,
+      pspReference: this.stripePaymentIntentId,
+      // https://docs.stripe.com/payments/paymentintents/lifecycle
+      message: "Payment intent succeeded",
+    });
+
+    return Response.json(typeSafeResponse, { status: 200 });
+  }
+
+  getTransactionEventReportVariables() {
+    // TODO: add implementation when handling Stripe webhook
+    throw new BaseError("Not implemented");
```

Comment:
I think it's already implemented in Stripe webhook, check
`apps/stripe/src/app/api/stripe/webhook/resolved-webhook-events.ts`